### PR TITLE
Exercise 8 and 9

### PR DESCRIPTION
**Pull request by lawli01**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/data/item.py [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+    @staticmethod
+    def from_str(label):
+        if label in ('Not Started'):
+            return ItemStatus.NOT_STARTED
+        elif label in ('In Progress'):
+            return ItemStatus.IN_PROGRESS
+        elif label in ('Complete'):
+            return ItemStatus.COMPLETE
+        else:
+            raise NotImplementedError
```
Comment (BenRamchandani) [inline todo_app/data/item.py:19]:
```suggestion
            raise NotImplementedError()
```
Throwing a class directly seems wrong, though I guess it works. Might also be worth providing an error message.

**Review by BenRamchandani — CHANGES_REQUESTED** [review]
Hey, this looks like a good start to both M8 and M9, but there's some more to do for both.

To finish M8 you'll need to get GitHub actions deploying to Heroku, and for M9 you'll need to update your tests to work with Mongo.

**Review by hugh-emerson — APPROVED** [review]
Your application is currently failing to load: 
http://todo-app-devopscourse.herokuapp.com/

Try checking the logs via the CLI or reach out to one of us if you need any extra help debugging it.

A guess at the issue would be environment variables in Heroku don't match the deployed version of the code, e.g.:
- Missing `MONGO_CONNECTION_STRING` or `MONGO_DATABASE_NAME`?
- Or you did set those in Heroku and deleted the Trello ones, but you haven't actually deployed the latest version of your code (since your pipeline only deploys from the main branch).

Approving PR because what you have here looks good.

File: .github/workflows/my-ci-pipeline.yml [hunk 2]
```diff
@@ -24,4 +24,45 @@ jobs:
...
+    name: Setup heroku login
+    run: |
+      cat >~/.netrc <<EOF
+        machine api.heroku.com
+          login ${{ secrets.HEROKU_EMAIL }}
+          password ${{ secrets.HEROKU_API_KEY }}
+        machine git.heroku.com
+          login ${{ secrets.HEROKU_EMAIL }}
+          password ${{ secrets.HEROKU_API_KEY }}
+      EOF
```
[the remark this answers is not in the document]
Comment (lawli01) [reply .github/workflows/my-ci-pipeline.yml:59]: I think I tried this and it didn't seem to work

**Pull request merged**